mypy_django_plugin/lib/helpers.py

```diff
@@ -43,8 +44,17 @@
     from mypy_django_plugin.django.context import DjangoContext
 
 
-def get_django_metadata(model_info: TypeInfo) -> Dict[str, Any]:
-    return model_info.metadata.setdefault("django", {})
+class DjangoTypeMetadata(TypedDict, total=False):
+    from_queryset_manager: str
+    reverse_managers: Dict[str, str]
+    baseform_bases: Dict[str, int]
+    manager_bases: Dict[str, int]
+    model_bases: Dict[str, int]
+    queryset_bases: Dict[str, int]
```

Comment:
I also think these different mappings are individually used. What I mean is that we use mypy's api to lookup a symbol(`sym`) instance where we mutate its metadata.

We always look for the same symbol instance, depending on if there's a model(`MODEL_CLASS_FULLNAME`), manager(`MANAGER_CLASS_FULLNAME`), etc

What I mean is that we won't add metadata for forms on the symbol instance of the base model.

I'm not totally sure if we could represent that in a nicer way. But I think this is still an improvement.

Something contextually bound like (warning, pseudo):

```python
T = TypeVar("T", Literal["baseform_bases"], Literal["manager_bases"], ...)
class DjangoTypeMetadata(dict[T, int]):
    ...

def get_django_metadata(context: T, info: TypeInfo) -> DjangoTypeMetadata[T]:
    ...
```

Is that understandable? That we'll never work with more than 1 key at a time from the typed dict you declared here.

Reply:
Hmm, seems simpler to create a second helper function like this:

```python
def get_django_metadata_bases(
    model_info: TypeInfo, key: Literal["baseform_bases", "manager_bases", "model_bases", "queryset_bases"]
) -> Dict[str, int]:
    return get_django_metadata(model_info).setdefault(key, {})
```

Reply:
I've pushed this change, I think it does something similar to what you were thinking?

Reply:
Yes, precisely